strawberry_django/auth/mutations.py

```diff
@@ -19,20 +23,39 @@
 
 @django_resolver
 def resolve_login(info: Info, username: str, password: str) -> AbstractBaseUser | None:
-    request = info.context.request
+    request = get_request(info)
     user = auth.authenticate(request, username=username, password=password)
+
     if user is not None:
-        auth.login(request, user)
+        try:
+            auth.login(request, user)
+        except AttributeError:
+            # ASGI in combo with websockets needs the channels login functionality.
+            # to ensure we're talking about channels, let's veriy that our
+            # request is actually channelsrequest
+            scope = request.consumer.scope  # type: ignore
+            async_to_sync(channels_auth.login)(scope, user)
+            # According to channels docs you must save the session
+            scope["session"].save()
         return user
+
+    # FIXME: Why call logout inside login??
```

Comment:
Indeed... I don't use this mutation in my projects, and I'm not the author of it.

IMO this seems wrong. Do you want to fix this `FIXME` by removing the `logout` call?

Reply:
@bellini666 I'll have a look at it, as this logout doesnt seem to actually do anything - which makes me suspect the actual logout mutation might not work.  I would appreciate some thoughts on #394 though.

Reply:
Seems to be fine to remove.  Logout mutation itself works fine.